### PR TITLE
Enable specifying the GET endpoint cache-control header via config

### LINKED ISSUE #56: Support for changing the Cache-Control header sent by the GET API
(followup to https://github.com/splitgraph/seafowl/issues/20)

The current default is not sending any cache-control headers at all, which means that the browser/CDNs will lean towards more caching. (from my basic testing, it seems like Cloudflare caches for 2h at least: https://developers.cloudflare.com/cache/about/edge-browser-cache-ttl/)

Possible settings:

  - how soon to revalidate with an If-None-Match (maxage?)
  - sending s-maxage as well as maxage (the former seems to influence CDNs) 

Some reference: https://jakearchibald.com/2016/caching-best-practices/

### PR DESCRIPTION
**Pull request by gruuya**
Closes #56

### CODE REVIEW

File: src/frontend/http.rs [hunk 1]
```diff
@@ -878,9 +896,16 @@ pub mod tests {
         #[case] maybe_headers: Option<Vec<(&'_ str, &'_ str)>>,
         #[values(None, Some("test_db"))] new_db: Option<&str>,
         #[values("", ".bin")] extension: &str,
+        #[values(None, Some("max-age=3600, public"))] cache_control: Option<&str>,
```
Comment (mildbyte) [inline src/frontend/http.rs:899]: Come to think about it, are these rstest cases multiplicative? i.e. if you add two values for cache-control here, do we end up running twice as many tests?

File: src/config/schema.rs [hunk 2]
```diff
@@ -289,6 +290,7 @@ impl Default for HttpFrontend {
             read_access: AccessSettings::Any,
             write_access: AccessSettings::Off,
             upload_data_max_length: 256,
+            cache_control: None,
```
Comment (mildbyte) [inline src/config/schema.rs:293]: TBH we could make `max-age=3600, public`, or some other suitable value, the default (either here or by emitting it in the initial seafowl.toml that we write out on first startup).

**Review by mildbyte — APPROVED** [review]

File: src/frontend/http.rs (hunk 1 above)
Comment (gruuya) [reply src/frontend/http.rs:899]: Yeah it's essentially a Cartesian product of the `case` and `values` sets, so we end up with all the previous tests run twice now.

```bash
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_1_query_in_path::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_2_query_in_body::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_3_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_1_None::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_2_Some___max_age3600_public___ ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_1_____::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_1_None ... ok
test frontend::http::tests::test_get_cached_no_etag::case_4_encoded_query_in_header::new_db_2_Some___test_db___::extension_2____bin__::cache_control_2_Some___max_age3600_public___ ... ok
```

File: src/config/schema.rs (hunk 2 above)
Comment (gruuya) [reply src/config/schema.rs]: I went with a default here (12hours, as 1 hour seemed to small?).

**Pull request merged**